lib/src/utils.dart

```diff
@@ -29,11 +29,17 @@ final Logger logger = new Logger('dependency_validator');
 
 String bulletItems(Iterable<String> items) => items.map((l) => '  * $l').join('\n');
 
-Iterable<File> listDartFilesIn(String dirPath) {
+Iterable<File> listDartFilesIn(String dirPath, List<String> excludedDirs) {
   if (!FileSystemEntity.isDirectorySync(dirPath)) return const [];
-  return new Directory(dirPath)
-      .listSync(recursive: true)
-      .where((entity) => entity is File && !entity.path.contains('/packages/') && entity.path.endsWith('.dart'));
+
+  return new Directory(dirPath).listSync(recursive: true).where((entity) {
+    if (entity is! File) return false;
+    if (entity.path.contains('/packages/')) return false;
+    if (!entity.path.endsWith('.dart')) return false;
+    if (excludedDirs.any(entity.path.startsWith)) return false;
```

Comment:
This could cause false positives if a directory is a substring of the excluded dir and the user forgets a trailing slash.

You can use [`path.isWithin`](https://www.dartdocs.org/documentation/path/1.5.1/path/isWithin.html) to handle this.

Reply:
whoa... Dart never ceases to amaze me